File: args.js
width = 42;

inputArgs = [
	[7,1,2],
	[2,2,0],

	[5,1,2],
	[2,2,0],

	[3,1,0],
	[2,2,0]
];

function getNewWidth(w, p, k, s) {
	return (w + p * 2 - k) / s + 1;
}

inputArgs.forEach(a => {
	var k = Number(a[0]);
	var s = Number(a[1]);
	var p = Number(a[2]);
	var o = width;
	var w = getNewWidth(width, p, k, s);
	width = Math.floor(w)
	console.log(o, k, s, p, width===w ? ' ' : '*', '=>', width)
})

File: args.py
import math

width = 15

inputArgs = [
	[3,1,1],
	[3,1,1],
	[3,1,1],
	[1,1,1]
]

def getNewWidth(w, p, k, s):
	return (w + p * 2.0 - k) / s + 1

for a in inputArgs:
	k = int(a[0])
	s = int(a[1])
	p = int(a[2])
	o = width
	w = getNewWidth(width, p, k, s)
	width = math.floor(w)
	if width==w:
		print(o, k, s, p, ' ', '=>', width)
	else:
		print(o, k, s, p, '*', '=>', width)
